src/Nethermind/Nethermind.Evm/TransactionProcessing/IReadOnlyTxProcessingScope.cs

```diff
@@ -9,5 +9,5 @@ namespace Nethermind.Evm.TransactionProcessing;
 public interface IReadOnlyTxProcessingScope : IDisposable
 {
     ITransactionProcessor TransactionProcessor { get; }
-    IWorldState WorldState { get; }
```

Comment:
Hmm... yea. This is a problem as the whole point of the scope is to represent a single IWorldState. So making it a WorldStateProvider changes that. So it probably should be called IReadOnlyTxProcessingEnv now.